From 323e5a44dea4198fdfb01380064c755fc0ec200e Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Sun, 30 Nov 2014 12:14:48 +0100
Subject: [PATCH] delta: Fix broken separator support

systemd-delta man page promises that multiple types of deltas will be
concatenated if they are listed with a comma as separator.  Replace
FOREACH_WORD() with FOREACH_WORD_SEPARATOR() to restore the functionality.

(cherry picked from commit d0a2e1c364320b862a6aae099afa106ea59bb84c)
---
 src/delta/delta.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/delta/delta.c b/src/delta/delta.c
index 9a2f17c58a..b73609d3f1 100644
--- a/src/delta/delta.c
+++ b/src/delta/delta.c
@@ -487,7 +487,7 @@ static int parse_flags(const char *flag_str, int flags) {
         const char *word, *state;
         size_t l;
 
-        FOREACH_WORD(word, l, flag_str, state) {
+        FOREACH_WORD_SEPARATOR(word, l, flag_str, ",", state) {
                 if (strneq("masked", word, l))
                         flags |= SHOW_MASKED;
                 else if (strneq ("equivalent", word, l))
